brian2/core/variables.py

```diff
@@ -844,6 +844,15 @@ def set_item(self, item, value, level=0, namespace=None):
         if variable.read_only:
             raise TypeError('Variable %s is read-only.' % self.name)
 
+        # Check whether the group allows writing to the variable (e.g. for
+        # synaptic variables, writing is only allowed after a connect)
+        try:
+            self.group.check_variable_write(variable)
+        except ReferenceError:
+            # Ignore problems with weakly referenced groups that don't exist
+            # anymore at this time (e.g. when doing neuron.axon.var = ...)
+            pass
```

Comment:
Is this not a problem that should result in an error? Or does it come up somewhere else?